docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5.1.1. [Target User Profile](#511-target-user-profile)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5.1.2. [Value Proposition](#512-value-proposition)<br>
+5.2. [User Stories](#52-user-stories)<br>
+5.3. [Use Cases](#53-use-cases)<br>
+5.4. [Non-Functional Requirements](#54-non-functional-requirements)<br>
+5.5. [Glossary](#55-glossary)<br>
+6. [<b>Appendix: Instructions for Manual Testing</b>](#6-appendix-instructions-for-manual-testing)<br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+--------------------------------------------------------------------------------------------------------------------
 
+## 1. Introduction
 
-## Product scope
-### Target user profile
+### 1.1. Overview
 
-{Describe the target user profile}
+### 1.2. Purpose
 
-### Value proposition
+### 1.3. Scope
+This documentation describes the software architecture and software design decisions for the implementation of Kaji. The intended audience of this document is the developers, designers, and software testers of Kaji.
 
-{Describe the value proposition: what problem does it solve?}
+--------------------------------------------------------------------------------------------------------------------
 
-## User Stories
+## 2. Setting Up
+
+### 2.1. Prerequisites
+* JDK 11
+* IntelliJ IDEA
+
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` → `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle.
+    1. Click `Configure` → `Project Defaults` → `Project Structure`.
+    2. Click `New…` and set it to the directory of the JDK.
+4. Click `Import Project` (or `Open or Import` in newer version of Intellij).
+5. Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+If asked, choose to `Open as Project` (not `Open as File`).
+7. Click `OK` to accept the default settings
+
+--------------------------------------------------------------------------------------------------------------------
+
+## 3. Design (Architecture)
+(Zeyu)
+
+![Architecture Diagram of Design Component](UML/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+The main class is `Kaji.java`. It is responsible for:
+
+* Initializing the components in the correct sequence, and connects them up with each other at app launch.
+* Shutting down the components and invoking cleanup methods where necessary at exit.
+
+The rest of the App consists of 8 components:
+* `Access`: Keeps track of the access level of the user.
+* `UI`: The UI of the App.
+* `Command`: Executes the different command types.
+* `Common`: Contains common classes.
```

Comment:
From the diagram and subsequent explanation, I am not sure I fully understand which of the following components are connected to the `Common` component. The `Common` component seems to be isolated from the other components.